### ISSUE TITLE
Lecture 7: Financial Markets Competition

### ISSUE THREAD

**Issue by luziusmeisser** [davidrpugh/sfi-complexity-mooc#3]
Inspired by _Leverage causes fat tails and clustered volatility_ by Thurner et al., we create an artificial stock market on which agents can trade with leverage. Hopefully, there will be wild dynamics despite the agents investing reasonably when considered individually. I would like to try this out without having to resort to noise traders, so it will only be the value investors that trade among themselves.

We would start by providing a base simulation with a few example agents, which the students can modify in order to create their favorite to submit to the competition. Providing example agents enables the students to immediately experiment with the interesting parameters instead of having to figure out the interfaces first. The agents could buy and sell stocks on an artificial market with incomplete information.

In a completely symmetric situation, no trade would happen at all. The incomplete information serves to break that symmetry. I think one could simply give the traders different (randomized) analyst opinions about the future dividends of each stock.

As a next step, I'll try to create such a prototype of such a model and see if it indeed leads to excess volatility.

**Comment by davidrpugh** [comment]
@kronrod 

This seems like a good approach.  It is unclear to me what you mean by incomplete, symmetric, and asymmetric information in this context.  Are you talking about valuations? If all agents have identical (symmetric) valuations of the same asset then there is no room for trade. However, if agents have different (asymmetric) valuations of the same asset then there is scope for trade.  Agent valuations are assumed private and hence information is incomplete? Can you confirm that this is what you have in mind?

Also we are going to have a lecture on the SFI stock market so it would be good to have this lecture reuse the market structure that we would build for that lecture.

Finally, Doyne is keen to leverage [ScalABM](https://github.com/ScalABM/markets-sandbox) which is a Scala, Akka based market microstructure platform that we are developing here at INET for running large scale market microstructure simulations.
I am not convinced that it is a good idea to incorporate cutting edge research tools into a MOOC whose target audience is advanced undergraduates...thoughts?

**Comment by luziusmeisser** [comment]
@davidrpugh 
I did not work out the all the details yet, so there is no guarantee it will add up. At the moment, I indeed envision the agents to have different information. They know the latest dividends of each company (which follow a random walk), but they each have their own analyst for estimations on future dividends. (In the background, we can per-calculate future dividends in advance, which form the basis for the slightly randomized analysts' estimates). Combined with leverage, this could lead to the desired volatility (although I am not convinced yet that it will work). Maybe there needs to be an additional mechanism to remove liquidity from the system (i.e. by letting firms issue new shares when there market cap is very high relative to shareholder equity) in order to provoke liquidity crises. I plan to run some experiments on this tomorrow.

**Comment by luziusmeisser** [comment]
@davidrpugh 
I agree regarding ScalABM. We should stick to the simplest possible tools that do the job and not switch back and forth between different systems. Unless there are good technical reasons to use ScalABM, I would prefer using Python. Also, there is always the risk with cutting edge technology that it won't work for a few student for some unexpected reason.

**Comment by davidrpugh** [comment]
@kronrod 

> They know the latest dividends of each company (which follow a random walk), but they each have their own analyst for estimations on future dividends. 

Agents having different rules for forecasting future prices/dividends is perfectly reasonable and would tie in nicely with lecture 2 on Cars Hommes' work.

I think it is important, for pedagogical reasons, to stay reasonably close to the _Leverage Causes Fat Tails and Clustered Volatility_ paper.  I think it is good for students to be able to see the connection between the maths in the paper and the simulations.  It is also nice for students to see that published academic work can be replicated...

**Comment by luziusmeisser** [comment]
@davidrpugh 

Regarding the paper: In the abstract, the authors claim "This is in contrast to previous explanations of fat tails and clustered volatility, which depended on ‘irrational behavior’, such as trend following." However, the noise trader in that model does not seem very rational to me. This is bothering me a little and that's why I deviated to a model without noise traders. The core point of the paper is that leverage can cause fat tails and clustered volatility. If we could replicate this finding in a more elegant setting, that would strengthen that finding more than just replicating everything 1:1. Also, I am not sure if the tournament is interesting enough if all the funds can decide on is demand.

Btw: A fun strategy in that setting might be to intentionally dump all stocks on the market at the right moment in order to trigger a sell-off and then buy them back cheaply a little later.

**Comment by luziusmeisser** [comment]
I just forked your repository to experiment on a "leverage" notebook. Before you start working on that topic, please let me know so we can share/merge my latest results. It could take a week or two before anything useful appears. :)

https://github.com/kronrod/sfi-complexity-mooc

**Comment by davidrpugh** [comment]
@kronrod 
Just make sure to do your work on a separate branch and then open a pull request against the `develop` branch on this repo.  In general I try to follow this [development model](http://nvie.com/posts/a-successful-git-branching-model/) on my projects.
